c_glib/arrow-glib/file-system.h

```diff
@@ -0,0 +1,236 @@
+
+GARROW_AVAILABLE_IN_1_0
+gboolean garrow_file_stats_is_file(GArrowFileStats *file_stats);
+GARROW_AVAILABLE_IN_1_0
+gboolean garrow_file_stats_is_directory(GArrowFileStats *file_stats);
+
+/* arrow::fs::FileSelector */
+
+#define GARROW_TYPE_FILE_SELECTOR (garrow_file_selector_get_type())
+G_DECLARE_DERIVABLE_TYPE(GArrowFileSelector,
+                         garrow_file_selector,
+                         GARROW,
+                         FILE_SELECTOR,
+                         GObject)
+struct _GArrowFileSelectorClass
+{
+  GObjectClass parent_class;
+};
+
+/* arrow::fs::FileSystem */
+
+#define GARROW_TYPE_FILE_SYSTEM (garrow_file_system_get_type())
+G_DECLARE_DERIVABLE_TYPE(GArrowFileSystem,
+                         garrow_file_system,
+                         GARROW,
+                         FILE_SYSTEM,
+                         GObject)
+struct _GArrowFileSystemClass
+{
+  GObjectClass parent_class;
+};
+
+GARROW_AVAILABLE_IN_1_0
+const gchar *garrow_file_system_get_type_name(GArrowFileSystem *file_system);
+
+GARROW_AVAILABLE_IN_1_0
+GArrowFileStats *garrow_file_system_get_target_stats_path(GArrowFileSystem *file_system,
+                                                          const gchar *path,
+                                                          GError **error);
+
+GARROW_AVAILABLE_IN_1_0
+GList *garrow_file_system_get_target_stats_paths(GArrowFileSystem *file_system,
+                                                 const gchar **paths,
+                                                 gsize n_paths,
+                                                 GError **error);
+
+GARROW_AVAILABLE_IN_1_0
+GList *garrow_file_system_get_target_stats_selector(GArrowFileSystem *file_system,
+                                                    GArrowFileSelector *file_selector,
+                                                    GError **error);
+
+GARROW_AVAILABLE_IN_1_0
+gboolean garrow_file_system_create_dir(GArrowFileSystem *file_system,
+                                       const gchar *path,
+                                       gboolean recursive,
+                                       GError **error);
+
+GARROW_AVAILABLE_IN_1_0
+gboolean garrow_file_system_delete_dir(GArrowFileSystem *file_system,
+                                       const gchar *path,
+                                       GError **error);
+
+GARROW_AVAILABLE_IN_1_0
+gboolean garrow_file_system_delete_dir_contents(GArrowFileSystem *file_system,
+                                                const gchar *path,
+                                                GError **error);
+
+GARROW_AVAILABLE_IN_1_0
+gboolean garrow_file_system_delete_file(GArrowFileSystem *file_system,
+                                        const gchar *path,
+                                        GError **error);
+
+GARROW_AVAILABLE_IN_1_0
+gboolean garrow_file_system_delete_files(GArrowFileSystem *file_system,
+                                        const gchar **paths,
```

Comment:
@kou Should we use `GList *` for the type of `paths` here because this function cannot be called from Ruby due to ArgumentError?

Reply:
`(array length=n_paths)` annotation is missing for `@paths`.
See `garrow_csv_read_options_set_null_values()` documentation for details.

Reply:
Resolved.  Thank you very much!